src/routes/committees/[shortName]/+page.server.ts

```diff
@@ -78,72 +88,71 @@ export const load: PageServerLoad = async ({ params }) => {
       },
     },
   });
-  if (!committee) {
-    throw error(404, "Committee not found");
-  }
+  const form = superValidate(committee, updateSchema);
   return {
     committee,
     positions: committee.positions,
     uniqueMemberCount: uniqueMembersInCommittee,
     numberOfMandates,
+    form,
   };
 };
 
 export const actions = {
   update: async ({ params, request, locals }) => {
-    const session = await locals.getSession();
-    await ctxAccessGuard(apiNames.COMMITTEE.UPDATE, session?.user);
     const formData = await request.formData();
-    const image = formData.get("image");
-    formData.delete("image"); // for serialization purposes
-    let newImageUploaded = false;
-    if (image && image instanceof File && image.size > 0) {
-      console.log(image);
-      const path = `committees/${params.shortName}.svg`;
-      try {
-        const putUrl = await fileHandler.getPresignedPutUrl(
-          session?.user,
-          "dev-material",
-          path,
-          true
-        );
-        await fetch(putUrl, {
-          method: "PUT",
-          body: image,
-        });
-        newImageUploaded = true;
-      } catch (e) {
-        return fail(500, {
-          error: (e as Error | undefined)?.message ?? "Unable to upload image",
-          data: Object.fromEntries(formData),
-        });
-      }
-    }
-    try {
-      await prisma.committee.update({
-        where: { shortName: params.shortName },
-        data: {
-          name: (formData.get("name") as string | null) ?? undefined,
-          description: (formData.get("description") as string | null) ?? undefined,
-          imageUrl: newImageUploaded
-            ? `minio/material/committees/${params.shortName}.svg?version=${new Date().getTime()}`
-            : undefined,
-        },
-      });
-      return {
-        success: true,
-        data: Object.fromEntries(formData),
-      };
-    } catch (e) {
-      if (e instanceof Prisma.PrismaClientKnownRequestError) {
-        if (e.code === "P2025" || e.code === "2016") {
-          return fail(404, { error: "Committee not found", data: Object.fromEntries(formData) });
+    const form = await superValidate(formData, updateSchema);
+    if (!form.valid) return fail(400, { form });
+    const session = await locals.getSession();
+    return withAccess(
+      apiNames.COMMITTEE.UPDATE,
+      session?.user,
+      async () => {
+        const image = formData.get("image");
+        if (
+          !image ||
+          !(image instanceof File) ||
+          image.size <= 0 ||
+          image.type !== "image/svg+xml"
+        ) {
+          return setError(form, "image", "Du måste ladda upp en korrekt bild");
```

Comment:
Could this error perhaps be more helpful? Is the svg requirement reflected in the frontend file upload. If not it should probably be reflected in this error.